### ISSUE TITLE
PYTHONHOME set to virtualenv throws Fatal Python error: init_fs_encoding: failed to get the Python codec of the filesystem encoding

### ISSUE THREAD

**Issue by JBlaschke** [LLNL/spack#32515]
### Steps to reproduce

Run 

```console
spack style
```

within a virtualenv. Ensure spack has set the `PYTHONHOME` variable to the virtualenv

### Error message

```
lib/spack/spack/installer.py:1961 ==> [2022-09-04-17:46:30.488898] py-pip: Executing phase: 'install'
lib/spack/spack/util/executable.py:197 ==> [2022-09-04-17:46:30.491269] '/home/blaschke/local/virtualenv/py3/bin/python3.8' '/tmp/blaschke/spack-stage/spack-stage-py-pip-22.1.2-6wrjgnq7ccpa44t5fwqzi22ryji3eiqz/pip-22.1.2-py3-none-any.whl/pip' '-vvv' '--no-input' '--no-cache-dir' '--disable-pip-version-check' 'install' '--no-deps' '--ignore-installed' '--no-build-isolation' '--no-warn-script-location' '--no-index' '--prefix=/home/blaschke/.spack/bootstrap/store/linux-ubuntu20.04-x86_64/gcc-12.1.0/py-pip-22.1.2-6wrjgnq7ccpa44t5fwqzi22ryji3eiqz' '/tmp/blaschke/spack-stage/spack-stage-py-pip-22.1.2-6wrjgnq7ccpa44t5fwqzi22ryji3eiqz/pip-22.1.2-py3-none-any.whl'
Python path configuration:
  PYTHONHOME = '/home/blaschke/local/virtualenv/py3'
  PYTHONPATH = '/home/blaschke/.spack/bootstrap/store/linux-ubuntu20.04-x86_64/gcc-12.1.0/py-pip-22.1.2-6wrjgnq7ccpa44t5fwqzi22ryji3eiqz/lib/python3.8/site-packages'
  program name = '/home/blaschke/local/virtualenv/py3/bin/python3.8'
  isolated = 0
  environment = 1
  user site = 1
  import site = 1
  sys._base_executable = '/home/blaschke/local/virtualenv/py3/bin/python3.8'
  sys.base_prefix = '/home/blaschke/local/virtualenv/py3'
  sys.base_exec_prefix = '/home/blaschke/local/virtualenv/py3'
  sys.executable = '/home/blaschke/local/virtualenv/py3/bin/python3.8'
  sys.prefix = '/home/blaschke/local/virtualenv/py3'
  sys.exec_prefix = '/home/blaschke/local/virtualenv/py3'
  sys.path = [
    '/home/blaschke/.spack/bootstrap/store/linux-ubuntu20.04-x86_64/gcc-12.1.0/py-pip-22.1.2-6wrjgnq7ccpa44t5fwqzi22ryji3eiqz/lib/python3.8/site-packages',
    '/home/blaschke/local/virtualenv/py3/lib/python38.zip',
    '/home/blaschke/local/virtualenv/py3/lib/python3.8',
    '/home/blaschke/local/virtualenv/py3/lib/python3.8/lib-dynload',
  ]
Fatal Python error: init_fs_encoding: failed to get the Python codec of the filesystem encoding
Python runtime state: core initialized
ModuleNotFoundError: No module named 'encodings'

Current thread 0x00007faf8b5d1740 (most recent call first):
<no Python frame>
```

Note that simply running:
```console
PYTHONHOME='/home/blaschke/local/virtualenv/py3' PYTHONPATH='/home/blaschke/.spack/bootstrap/store/linux-ubuntu20.04-x86_64/gcc-12.1.0/py-pip-22.1.2-6w
rjgnq7ccpa44t5fwqzi22ryji3eiqz/lib/python3.8/site-packages' python
```
Throws the same error.

### Information on your system

```
* **Spack:** 0.19.0.dev0 (b68e82bbd0fc668f22adcb78af941bc3f1a2ee5b)
* **Python:** 3.8.10
* **Platform:** linux-ubuntu20.04-icelake
* **Concretizer:** clingo
```

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have searched the issues of this repo and believe this is not a duplicate
- [X] I have run the failing commands in debug mode and reported the output

**Comment by cessenat** [comment]
Got the issue too

**Comment by alalazo** [comment]
> Ensure spack has set the PYTHONHOME variable to the virtualenv

Cannot reproduce as of:

* **Spack:** 0.20.0.dev0 (a7b2196eab54de4bb248f1706a1a69fbd1a9758f)
* **Python:** 3.8.10
* **Platform:** linux-ubuntu20.04-icelake
* **Concretizer:** clingo

Possibly fixed in #33956
